ui/components/src/LineChart/LineChart.tsx

```diff
@@ -100,7 +101,18 @@ export function LineChart({ height, data, unit, grid, legend, visualMap, onDataZ
     enableDataZoom(chartRef.current);
   }
 
+  const handleOnClick = () => {
+    if (pinTooltip === false) {
+      setPinTooltip(true);
+    } else {
+      setPinTooltip(false);
+    }
+  };
+
   const handleOnDoubleClick = () => {
+    if (pinTooltip === true) {
+      setPinTooltip(false);
```

Comment:
Same nit here, should use the callback form since you're relying on the current state value.